### PR TITLE
Add gradle project

### PR DESCRIPTION
**Pull request by Joaquimmnetto**
## Description
Added simple gradle 7.0 project structure based on [this](https://github.com/jenkins-infra/pipeline-library/blob/master/pom.xml) maven project done by the Jenkins team for a library of theirs. This project shouldn't generate any artifact, just check for compilation and (in future) test errors.

## Changes
* ![ADD] Gradle project files

[ADD]:https://resources.atlas.wooga.com/icons/icon_add.svg "Add"

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.idea
+*.iml
```
Comment (Larusso) [inline .gitignore:2]: I personally keep these in a global ignore list on my systems. It's not bad you just have to imagine how the file would look like if we would add all kinds of project files (vscode, sublime, eclipse, ...)

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ */
+
+plugins {
+    id 'groovy'
+    id 'maven-publish'
+}
+
+group = 'io.jenkins.infra'
```
Comment (Larusso) [inline build.gradle:10]: I guess it makes sense to use a group name with our values like 'com.wooga' or 'net.wooga'.

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * This file was generated by the Gradle 'init' task.
+ */
+
+plugins {
+    id 'groovy'
+    id 'maven-publish'
+}
+
+group = 'io.jenkins.infra'
+version = '0.0.1'
```
Comment (Larusso) [inline build.gradle:11]: Our library is versioned. Check the tags.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    main {
+        groovy {
+            srcDirs = ['src/', 'vars/']
+        }
+    }
+    // test {
+    //     groovy {
+    //         srcDirs = ['test']
+    //     }
+    // }
```
Comment (Larusso) [inline build.gradle:37]: Why is this not added?

**Review by Larusso — CHANGES_REQUESTED** [review]
That is cool. I hoped that it could be as simple as that. Can you remove the changes to the buildsteps for this PR or are they needed to compile the project in Gradle?

File: build.gradle (hunk 4 above)
Comment (Joaquimmnetto) [reply build.gradle:37]: Cause there is not tests ;)

File: .gitignore (hunk 1 above)
Comment (Joaquimmnetto) [reply .gitignore]: Alright, removed :)

File: build.gradle (hunk 2 above)
Comment (Joaquimmnetto) [reply build.gradle]: Yeah, this was a copy leftover. Changed it.

File: build.gradle (hunk 3 above)
Comment (Joaquimmnetto) [reply build.gradle]: Changed version to the 1.20.2 (current version is 1.20.1, so I did a minor upgrade).

Comment (Larusso) [reply build.gradle]: hmm. You could just use the same version as the version indicates the API not the meta around it.

File: build.gradle (hunk 4 above)
Comment (Larusso) [reply build.gradle]: Gotcha!

File: build.gradle (hunk 3 above)
Comment (Joaquimmnetto) [reply build.gradle]: Done.

File: src/net/wooga/jenkins/pipeline/BuildUtils.groovy [hunk 5]
```diff
@@ -1,6 +1,6 @@
 package net.wooga.jenkins.pipeline
 
-import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+import com.cloudbees.groovy.cps.NonCPS
```
Comment (Larusso) [inline src/net/wooga/jenkins/pipeline/BuildUtils.groovy:3]: These are the changes I meant. Are these needed for this PR? If yes all is fine.

File: src/net/wooga/jenkins/pipeline/TestHelper.groovy [hunk 6]
```diff
@@ -1,6 +1,5 @@
 package net.wooga.jenkins.pipeline
-
-import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+//import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
```
Comment (Larusso) [inline src/net/wooga/jenkins/pipeline/TestHelper.groovy:2]: and here

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+# Avoid ignoring Gradle wrapper jar file (.jar files are usually ignored)
+!gradle-wrapper.jar
+
+# Cache of project
+.gradletasknamecache
+
+# # Work around https://youtrack.jetbrains.com/issue/IDEA-116898
+# gradle/wrapper/gradle-wrapper.properties
+userHome/
```
Comment (Larusso) [inline .gitignore:18]: Can you please add the missing newline

File: build.gradle [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    implementation 'org.codehaus.groovy:groovy-all:3.0.8'
+}
+
+sourceSets {
+    main {
+        groovy {
+            srcDirs = ['src/', 'vars/']
+        }
+    }
+}
```
Comment (Larusso) [inline build.gradle:33]: Can you please add the missing newline

**Review by Larusso — CHANGES_REQUESTED** [review]

File: src/net/wooga/jenkins/pipeline/BuildUtils.groovy (hunk 5 above)
Comment (Joaquimmnetto) [reply src/net/wooga/jenkins/pipeline/BuildUtils.groovy:3]: The imports changes are needed as these imports were not working and not being used anymore. The `@NonCPS` annotation import is needed in order for it to compile. I guess that at a jenkins level, it should be already in the environment, as this was working before the import was added.

File: src/net/wooga/jenkins/pipeline/TestHelper.groovy (hunk 6 above)
Comment (Joaquimmnetto) [reply src/net/wooga/jenkins/pipeline/TestHelper.groovy:2]: This import wasnt being used by the script, and it didn't exist in the refered library. However, I'll remove the comments.

Comment (Joaquimmnetto) [reply src/net/wooga/jenkins/pipeline/TestHelper.groovy]: Done.

File: .gitignore (hunk 7 above)
Comment (Joaquimmnetto) [reply .gitignore]: Done

File: build.gradle (hunk 8 above)
Comment (Joaquimmnetto) [reply build.gradle]: Done

File: src/net/wooga/jenkins/pipeline/BuildUtils.groovy (hunk 5 above)
Comment (Larusso) [reply src/net/wooga/jenkins/pipeline/BuildUtils.groovy:3]: OK very well.

**Review by Larusso — APPROVED** [review]

**Pull request merged**